build.js

```diff
@@ -0,0 +1,167 @@
+const execSync = require('child_process').execSync;
+const basePath = process.argv[2];
+const webcomponent = process.argv[3];
+let fs = require('fs');
+let path = require('path');
+
+class WebComponentBuild {
+	constructor(path, webcomponent) {
+		this.path = path;
+		this.webcomponent = webcomponent;
+		this.srcFolder = `${this.path}/src`;
+		this.distFolder = `${this.path}/dist`;
+	}
+
+	execute() {
+		this.__cleanDistFolder();
+		this.__copyNonVlSrcToDist();
+		fs.readdirSync(this.srcFolder).forEach(file => {
+			if (file.startsWith("vl-")) {
+				this.__build(path.resolve(this.srcFolder, file));
+			}
+		});
+		this.__commit();
+	}
+	
+	__cleanDistFolder() {
+		fs.rmdirSync(this.distFolder, { recursive: true });
+		fs.mkdirSync(this.distFolder);
+	}
+
+	__copyNonVlSrcToDist() {
+		copyFilesTo(this.srcFolder, this.distFolder, file => !file.startsWith("vl-"));
+	}
+	
+	__build(file) {
+		this.__buildEs6(file);
+		this.__buildEs6Min(file);
+		this.__buildNode(file);
+	}
+	
+	__buildEs6(file) {
+		const es6BuildFile = `${this.distFolder}/${fileNameWithoutExtension(file)}.js`;
+		copy(file, es6BuildFile);
+		this.__maakStyleImportAbsoluutNaarDist(es6BuildFile);
+		this.__maakLibImportsAbsoluut(es6BuildFile);
+		this.__maakVlSrcImportsAbsoluut(es6BuildFile);
+		this.__maakSrcImportsAbsoluut(es6BuildFile);
+	}
+	
+	__buildEs6Min(file) {
+		const es6MinBuildFile = `${this.distFolder}/${fileNameWithoutExtension(file)}.min.js`;
+		copy(file, es6MinBuildFile);
+		this.__vervangWebcomponentenImportsDoorMinifiedImports(es6MinBuildFile);
+		this.__maakLibImportsAbsoluut(es6MinBuildFile);
+		this.__maakVlSrcImportsAbsoluut(es6MinBuildFile);
+		this.__maakSrcImportsAbsoluut(es6MinBuildFile);
+		this.__inlineCss(es6MinBuildFile);
+		this.__minify(es6MinBuildFile);
+	}
+
+	__buildNode(file) {
+		const nodeBuildFile = `${this.distFolder}/${fileNameWithoutExtension(file)}.src.js`;
+		copy(file, nodeBuildFile);
+		this.__vervangWebcomponentenImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangThirdPartyImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangLocalLibImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangLocalVlSrcImportsDoorCommonJsImports(nodeBuildFile);
+		this.__vervangLocalSrcImportsDoorCommonJsImports(nodeBuildFile);
```

Comment:
Technisch gezien zijn het nog ESM imports (`import`) en geen common JS imports (`require`).
Ik vind het daardoor verwarrend dat we altijd spreken over de commonjs build. Ik denk dat het eerder een ESM build is die toevallig door het build systeem van een van onze afnemers kan gebruikt worden in een commonjs omgeving omdat we de absolute paden veranderen naar relatieve paden die door een import resolver in een build van een afnemer naar iets zinnigs worden omgevormd.
Correct me if I'm wrong, @coemans?
Is een betere naam in dat geval niet de "relative import" build en `vervangXYZImportsDoorRelativeImports`?

Reply:
Bij relatieve en absolute paden denk ik automatisch aan ES6 imports:
```JavaScript
import './relative.js';
import '/absolute.js';
```
Dus voorstel vind ik niet beter, maar sta uiteraard open voor andere voorstellen.

Reply:
Maar een CommonJS import is `import "vl-ui-fliefloefla"` toch al zeker niet aangezien er geen `require` aan te pas komt, dus dat lijkt me sowieso een foute benaming?

`import "module-name"` staat gewoon tussen de lijst met andere imports op https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import. Dat er een resolver nodig is om het pad te kunnen resolven naar een concrete file wil toch niet zeggen dat het geen ESM import is? Het importpad is gewoon niet native door de browser ondersteund aangezien er een resolver tussen moet zitten.
Lijkt me daarom dat het "relatief" noemen niet fout is, aangezien het relatief is aan het pad dat de resolver gebruikt. Of "esm-by-module-name" of iets dergelijks, maar dat is dan weer aan de lange kant :)

Reply:
aangepast naar "...RelatieveImports"